src/nfts/saga.ts

```diff
@@ -0,0 +1,54 @@
+import { call, put, select, spawn, takeLeading } from 'redux-saga/effects'
+import { fetchNfts, fetchNftsCompleted, fetchNftsFailed } from 'src/nfts/slice'
+import { getFeatureGate } from 'src/statsig'
+import { StatsigFeatureGates } from 'src/statsig/types'
+import Logger from 'src/utils/Logger'
+import { safely } from 'src/utils/safely'
+import { Actions } from 'src/web3/actions'
+import networkConfig from 'src/web3/networkConfig'
+import { walletAddressSelector } from 'src/web3/selectors'
+
+const TAG = 'NftsSaga'
+
+export function* handleFetchNfts() {
+  const showNftsInApp = getFeatureGate(StatsigFeatureGates.SHOW_IN_APP_NFT_GALLERY)
+  if (!showNftsInApp) {
+    Logger.debug(TAG, 'Feature gate not enabled, skipping NFTs list fetch')
+    return
+  }
+
+  const address = yield select(walletAddressSelector)
+  if (!address) {
+    Logger.debug(TAG, 'Wallet address not found, skipping NFTs list fetch')
+    return
+  }
+
+  const url = `${networkConfig.getNftsByOwnerAddressUrl}?address=${address}`
+
+  try {
+    const response = yield call(fetch, url, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+    })
+    if (!response.ok) {
+      throw new Error('Could not fetch NFTs')
```

Comment:
nit: This is fine, but maybe we can include some data of why it failed? At least the status code of the response?
I think it will make it easier for us if we have to debug.

Reply:
fixed in https://github.com/valora-inc/wallet/pull/3902/commits/4573c5e25a234bce0dfe27bf72e1ff1e9cc12e0d!